### PR TITLE
Fix a crash on examining the bulletin board under certain conditions

### PR DESCRIPTION
**Pull request by AMurkin**
#### Summary
```SUMMARY: Bugfixes "Fix a crash on examining the bulletin board under certain conditions"```

#### Purpose of change
Fixes crash mentioned in the https://github.com/CleverRaven/Cataclysm-DDA/issues/31105#issuecomment-498416633.

#### Describe the solution
Band-aid fix. Early return if cur_key_list is empty.
Have no idea what's happening.

### CODE REVIEW

File: src/mission_companion.cpp [hunk 1]
```diff
@@ -429,6 +429,11 @@ bool talk_function::display_and_choose_opts( mission_data &mission_key, const tr
 
     reset_cur_key_list();
 
+    if( cur_key_list.empty() ) {
+        popup( _( "There are no missions at this colony.  Press Spacebar..." ) );
```
Comment (ZhilkinSerg) [inline src/mission_companion.cpp]: Maybe reword to "Press any key..."?

Comment (AMurkin) [reply src/mission_companion.cpp]: This is the same message as at the beginning of the function.
https://github.com/CleverRaven/Cataclysm-DDA/blob/bcb6e0ec90e9d1e10f14bcd7b6e68c854b485c34/src/mission_companion.cpp#L368
IMO, better to leave it as it is to save translations.
